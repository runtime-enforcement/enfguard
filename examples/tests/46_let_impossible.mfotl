LET G1(x)+- = A(x) IN
ALWAYS (FORALL x. G1(x) AND NOT G1(x))