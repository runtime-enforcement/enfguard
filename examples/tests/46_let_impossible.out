The formula
 LET G1(x:int)+- = A(x:int) IN □[0s,∞) (∀x:int. ¬G1(x:int) ∧ G1(x:int))
is not enforceable. To make it enforceable, you would need to
 at the same time
 * make A(x:int) CauSup, but this is impossible (A(x:int) is never both causable and suppressable)
 * make LET G1(x:int)+- = A(x:int) IN □[0s,∞) (∀x:int. G1(x:int) ∧ ¬G1(x:int)) causable, but this is impossible (for causability x:int must be past-guarded)
